lib/plugins/aws/invoke-local/index.js

```diff
@@ -460,7 +460,13 @@ class AwsInvokeLocal {
     // Java zip/jar implementaion doesn't preserve file permissions in a zip file.
     // https://bugs.openjdk.java.net/browse/JDK-6194856
     // So we follow Java's way in java.util.zip.ZipEntry#isDirectory() to detect directory.
-    await decompress(artifact, destination, { filter: (file) => !file.path.endsWith('/') });
+    const artifactFiles = await decompress(artifact, destination, { filter: (file) => !file.path.endsWith('/') });
+    for (const file of artifactFiles) {
+      // We need to make sure there is executable permission for rust builds
```

Comment:
Let's provide a more in-depth explanation for this workaround, please see the example of explanation above for the Java-related workaround

Reply:
Please check https://github.com/serverless/serverless/pull/10891/commits/2a9e6b07ca4d96cf9e26dd24af4b02df23eb8f74